core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/common/decorator/SelectionSetSorter.java

```diff
@@ -0,0 +1,28 @@
+package ai.timefold.solver.core.impl.heuristic.selector.common.decorator;
+
+import java.util.Set;
+import java.util.SortedSet;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+
+import org.jspecify.annotations.NullMarked;
+
+/**
+ * Decides the order of a {@link Set} of selection values.
+ *
+ * @param <Solution_> the solution type, the class with the {@link PlanningSolution} annotation
+ * @param <T> the selection type
+ */
+@NullMarked
+@FunctionalInterface
+public interface SelectionSetSorter<Solution_, T> {
```

Comment:
Not sure why this needs to exist.
Why not have `SelectionSorter` accept and return a `Collection`?
Strangely, `SelectionSorter` doesn't return _anything_...

Also, we have the comparator factory class now. Why do we even have this type anymore?

Reply:
> Why not have SelectionSorter accept and return a Collection?

We can't do that because some selectors depend on the collection type being a `List`. Operations such as `listIterator()` cannot be performed with the base `Collection` type.